ast/src/parsed/display.rs

```diff
@@ -621,6 +623,38 @@ impl<E: Display> EnumDeclaration<E> {
     }
 }
 
+impl<E: Display> Display for TraitImplementation<E> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        let type_vars = if self.type_scheme.vars.is_empty() {
+            Default::default()
+        } else {
+            format!("<{}>", self.type_scheme.vars)
+        };
+
+        let trait_vars = if self.type_scheme.types.is_empty() {
+            Default::default()
+        } else {
+            let formatted_elements: Vec<String> = self
+                .type_scheme
+                .types
+                .iter()
+                .map(|t| format!("{t}"))
+                .collect();
+            format!("<{}>", formatted_elements.join(", "))
```

Comment:
Why not (using itertools)
```suggestion
            format!("<{}>", self
                .type_scheme
                .types
                .format(", "))
```